src/WorldStorage/WSSAnvil.cpp

```diff
@@ -1474,19 +1474,119 @@ cBlockEntity * cWSSAnvil::LoadSignFromNBT(const cParsedNBT & a_NBT, int a_TagIdx
+		{ "EnderDragon", &cWSSAnvil::LoadEnderDragonFromNBT },
+		{ "minecraft:ender_dragon", &cWSSAnvil::LoadEnderDragonFromNBT },
+		{ "Enderman", &cWSSAnvil::LoadEndermanFromNBT },
+		{ "minecraft:enderman", &cWSSAnvil::LoadEndermanFromNBT },
+		{ "Ghast", &cWSSAnvil::LoadGhastFromNBT },
+		{ "minecraft:ghast", &cWSSAnvil::LoadGhastFromNBT },
+		{ "Giant", &cWSSAnvil::LoadGiantFromNBT },
+		{ "minecraft:giant", &cWSSAnvil::LoadGiantFromNBT },
+		{ "Guardian", &cWSSAnvil::LoadGuardianFromNBT },
+		{ "minecraft:guardian", &cWSSAnvil::LoadGuardianFromNBT },
+		{ "Horse", &cWSSAnvil::LoadHorseFromNBT },
+		{ "minecraft:horse", &cWSSAnvil::LoadHorseFromNBT },
+		{ "Villager", &cWSSAnvil::LoadVillagerFromNBT },
+		{ "minecraft:villager", &cWSSAnvil::LoadVillagerFromNBT },
+		{ "VillagerGolem", &cWSSAnvil::LoadIronGolemFromNBT },
+		{ "minecraft:villager_golem", &cWSSAnvil::LoadIronGolemFromNBT },
+		{ "LavaSlime", &cWSSAnvil::LoadMagmaCubeFromNBT },
+		{ "minecraft:magma_cube", &cWSSAnvil::LoadMagmaCubeFromNBT },
+		{ "MushroomCow", &cWSSAnvil::LoadMooshroomFromNBT },
+		{ "minecraft:mooshroom", &cWSSAnvil::LoadMooshroomFromNBT },
+		{ "Ozelot", &cWSSAnvil::LoadOcelotFromNBT },
+		{ "minecraft:ocelot", &cWSSAnvil::LoadOcelotFromNBT },
+		{ "Pig", &cWSSAnvil::LoadPigFromNBT },
+		{ "minecraft:pig", &cWSSAnvil::LoadPigFromNBT },
+		{ "Rabbit", &cWSSAnvil::LoadRabbitFromNBT },
+		{ "minecraft:rabbit", &cWSSAnvil::LoadRabbitFromNBT },
+		{ "Sheep", &cWSSAnvil::LoadSheepFromNBT },
+		{ "minecraft:sheep", &cWSSAnvil::LoadSheepFromNBT },
+		{ "Silverfish", &cWSSAnvil::LoadSilverfishFromNBT },
+		{ "minecraft:silverfish", &cWSSAnvil::LoadSilverfishFromNBT },
+		{ "Skeleton", &cWSSAnvil::LoadSkeletonFromNBT },
+		{ "minecraft:skeleton", &cWSSAnvil::LoadSkeletonFromNBT },
+		{ "Slime", &cWSSAnvil::LoadSlimeFromNBT },
+		{ "minecraft:slime", &cWSSAnvil::LoadSlimeFromNBT },
+		{ "SnowMan", &cWSSAnvil::LoadSnowGolemFromNBT },
+		{ "minecraft:snowman", &cWSSAnvil::LoadSnowGolemFromNBT },
+		{ "Spider", &cWSSAnvil::LoadSpiderFromNBT },
+		{ "minecraft:spider", &cWSSAnvil::LoadSpiderFromNBT },
+		{ "Squid", &cWSSAnvil::LoadSquidFromNBT },
+		{ "minecraft:squid", &cWSSAnvil::LoadSquidFromNBT },
+		{ "Witch", &cWSSAnvil::LoadWitchFromNBT },
+		{ "minecraft:witch", &cWSSAnvil::LoadWitchFromNBT },
+		{ "WitherBoss", &cWSSAnvil::LoadWitherFromNBT },
+		{ "minecraft:wither", &cWSSAnvil::LoadWitherFromNBT },
+		{ "Wolf", &cWSSAnvil::LoadWolfFromNBT },
+		{ "minecraft:wolf", &cWSSAnvil::LoadWolfFromNBT },
+		{ "Zombie", &cWSSAnvil::LoadZombieFromNBT },
+		{ "minecraft:zombie", &cWSSAnvil::LoadZombieFromNBT },
+		{ "PigZombie", &cWSSAnvil::LoadPigZombieFromNBT },
+		{ "minecraft:zombie_pigman", &cWSSAnvil::LoadPigZombieFromNBT },
+	};
+
+	auto it = EntityTypeToFunction.find(AString(a_IDTag, a_IDTagLength));
+	if (it != EntityTypeToFunction.end())
+	{
+		(this->*it->second)(a_Entities, a_NBT, a_EntityTagIdx);
+	}
+
+	if (strncmp(a_IDTag, "Minecart", a_IDTagLength) == 0)
```

Comment:
You could add a new `LoadTypedMinecart` function that would handle these, and have everything in the map.